### PR TITLE
#1038 - Update documentation for ALPS.

### PR DESCRIPTION
**Pull request by gregturn**

### CODE REVIEW

**Comment by gregturn** [comment]
With #1064 merged to master, this PR has simplified to just adding test cases and documentation improvements.

File: src/main/java/org/springframework/hateoas/mediatype/alps/Alps.java [hunk 1]
```diff
@@ -38,14 +40,23 @@
...
+@JsonPropertyOrder({ "version", "doc", "descriptor" })
 @JsonInclude(JsonInclude.Include.NON_NULL)
 public class Alps {
 
-	private final String version = "1.0";
+	private final String version;
 	private final Doc doc;
 	private final List<Descriptor> descriptor;
 
+	@JsonCreator
```
Comment (odrotbohm) [inline src/main/java/org/springframework/hateoas/mediatype/alps/Alps.java]: Can you elaborate on why we need the creators? So far the classes have been designed to only be used for serialization and object creation is handled via builders.

File: src/main/java/org/springframework/hateoas/mediatype/alps/Descriptor.java [hunk 2]
```diff
@@ -33,14 +35,32 @@
...
-	private final String id, href, name;
+	private final String id;
+	private final String href;
+	private final String name;
 	private final Doc doc;
 	private final Type type;
 	private final Ext ext;
 	private final String rt;
 	private final List<Descriptor> descriptor;
+
```
Comment (odrotbohm) [inline src/main/java/org/springframework/hateoas/mediatype/alps/Descriptor.java]: See above question.

File: src/main/java/org/springframework/hateoas/mediatype/alps/Alps.java (hunk 1 above)
Comment (gregturn) [reply src/main/java/org/springframework/hateoas/mediatype/alps/Alps.java]: It makes deserialization work which seemed necessary for using WebTestClient. 

What’s wrong with making deserialization operational? We can make that constructor private.

Comment (gregturn) [reply src/main/java/org/springframework/hateoas/mediatype/alps/Alps.java]: I think I had test cases earlier that tapped deserialization, but I clearly changed them to use `jsonPath` in the MockMVC and WebTestClient test cases. I can remove the `@JsonCreator` if you'd like.

**Comment by gregturn** [comment]
Resolved via https://github.com/spring-projects/spring-hateoas/commit/f15740fb25f7babb34f423e9a54c331cb213e43a.

**Pull request closed without merging**